### PR TITLE
Adding bushslicer test suite.

### PR DESCRIPTION
**Pull request by kwoodson**
Adding a periodic job for bushslicer test suite.

This will require an image be built and pushed for bushslicer.

### CODE REVIEW

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 1]
```diff
@@ -383,3 +383,56 @@ periodics:
...
+        value: azure
+      - name: CONFIG_SPEC
+        valueFrom:
+          configMapKeyRef:
+            key: openshift-openshift-azure-master.yaml
+            name: ci-operator-configs
+      - name: JOB_NAME_SAFE
+        value: e2e-azure
+      - name: TEST_IMAGE
+        value: quay.io/kwoodson/bushslicer
```
Comment (kwoodson) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: Required update.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 2]
```diff
@@ -383,3 +383,56 @@ periodics:
...
+      - name: TEST_COMMAND
+        value: "/usr/bin/scl enable rh-git29 rh-ror50 -- cucumber features/cli/build.feature:5"
```
Comment (cjryan) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: In the near future, we'll want to update this with the full list of testcases to run. For now, this is fine as a place holder. Thanks!

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 3]
```diff
@@ -126,9 +132,24 @@ objects:
 
         # test specific code
         # don't let clients impact the global kubeconfig
-        mkdir -p /go/src/github.com/openshift/
-        cp -r /tmp/shared/openshift-azure /go/src/github.com/openshift/
-        cd /go/src/github.com/openshift/openshift-azure/
+        if [[ "${TEST_IMAGE}" == *"/bushslicer"* ]]; then
+          TEST_NODE_OPTS="etcd:master:node"
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Is it possible to handle all of these steps in a script in the bushslicer repo and then call that script using `TEST_COMMAND`?

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 4]
```diff
@@ -91,6 +91,12 @@ objects:
         value: /tmp/shared/home
       - name: TEST_IN_PRODUCTION
         value: ${TEST_IN_PRODUCTION}
+      - name: RESOURCEGROUP
+        value: e2e-${RESOURCEGROUP_SUFFIX}
+      - name: TEST_IMAGE
+        value: ${TEST_IMAGE}
+      - name: TEST_COMMAND
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Not sure you need this. TEST_COMMAND should already work as is, no?

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 1 above)
Comment (kargakis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: @droslean was planning to build a bushslicer image that will be served by our CI registry, maybe we can converge and use that.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 5]
```diff
@@ -126,9 +132,24 @@ objects:
 
         # test specific code
         # don't let clients impact the global kubeconfig
-        mkdir -p /go/src/github.com/openshift/
-        cp -r /tmp/shared/openshift-azure /go/src/github.com/openshift/
-        cd /go/src/github.com/openshift/openshift-azure/
+        if [[ "${TEST_IMAGE}" == *"/bushslicer"* ]]; then
+          TEST_NODE_OPTS="etcd:master:node"
+          TEST_CLUSTER="${RESOURCEGROUP}.eastus.cloudapp.azure.com"
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Let's avoid hardcoding eastus here, eventually we should not be setting AZURE_REGION and have the test pick one at random (we already do that in the upgrade test). Can't you read fqdn from containerservice.yaml in `/tmp/shared/openshift-azure`?

Comment (kargakis) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: If you are going to move this into a script in the bushslicer repo, then expect in the script that TEST_CLUSTER is already set and set it here before running TEST_COMMAND.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 6]
```diff
@@ -126,9 +132,24 @@ objects:
...
+          export TESTENV="BUSHSLICER_V3"
+          export BUSHSLICER_DEFAULT_ENVIRONMENT=online_starter
+          export BUSHSLICER_DEBUG_AFTER_FAIL=1
+          export OPENSHIFT_ENV_ONLINE_STARTER_HOSTS="${TEST_CLUSTER}:${TEST_NODE_OPTS}"
+          # login to cluster as customer-cluster-admin
+          PASSWD=$(awk '/customerAdminPasswd:/{ print $2 }' /tmp/shared/openshift-azure/_data/containerservice.yaml)
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: LIkewise, if this is going to be a standalone script, expect PASSWD to be set in the script.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 7]
```diff
@@ -126,9 +132,24 @@ objects:
...
+          oc login -u customer-cluster-admin -p "${PASSWD}" --server=https://${TEST_CLUSTER} --insecure-skip-tls-verify=true
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: I would also make the username configurable in the script.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 1 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: That's the plan.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 4 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: I thought it might be good to be explicit.  I can remove it if you'd prefer.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 8]
```diff
@@ -383,3 +383,56 @@ periodics:
     - configMap:
         name: prow-job-cluster-launch-e2e-azure-conformance
       name: job-definition
+- agent: kubernetes
+  cron: ""
+  decorate: true
+  interval: 24h
```
Comment (kwoodson) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: maybe longer duration?

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 9]
```diff
@@ -126,9 +132,24 @@ objects:
...
         # don't let clients impact the global kubeconfig
-        mkdir -p /go/src/github.com/openshift/
-        cp -r /tmp/shared/openshift-azure /go/src/github.com/openshift/
-        cd /go/src/github.com/openshift/openshift-azure/
+        if [[ "${TEST_IMAGE}" == *"/bushslicer"* ]]; then
+          TEST_NODE_OPTS="etcd:master:node"
+          TEST_CLUSTER="${RESOURCEGROUP}.eastus.cloudapp.azure.com"
+          export TESTENV="BUSHSLICER_V3"
+          export BUSHSLICER_DEFAULT_ENVIRONMENT=online_starter
+          export BUSHSLICER_DEBUG_AFTER_FAIL=1
```
Comment (cjryan) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Since we're not doing active development, and this is in ci/cd, we can remove the DEBUG_AFTER_FAIL line here, or set it to 0.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 5 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: I will fix the region bits.  I'm not sure the extra script will go into the bushslicer repository.  Since this is pretty specific to our use case.  Would placing these variables at the top in the parameters list be acceptable?

**Comment by kargakis** [comment]
Yes that is probably fine for now

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 3 above)
Comment (mjudeikis) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: ok, this is one of those things we need to consider very well. 
1. To move to bushslicer repo means - one more repo to interact with, where we dont have much power. So any time we need to merge anything or change anything - we will have to get approved, etc. It might be right place, but don't think it worth the pain.
2. We can move this to openshift-azure repo and move scripts as part of `/tmp/shared` with right target calls... 
3. We can host scripts somewhere outside and pull them in (not the biggest fan of this).
4. We keep them at a minimum in the template, as there is a small change we will be able to consolidate on "no templates at all" any time soon.

@kargakis lets discuss this in the morning :)

Comment (stevekuznetsov) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Strongly do not prefer changes to openshift/release -- the jobs here should be as static as possible.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 2 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: @cjryan, please supply the full list.  I think it's time.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 4 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: will remove.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 5 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: I attempted to make these variables parameters but ran into an issue with it not being available in the test environment:
In the parameters:
```
- name: BUSHSLICER_DEFAULT_ENVIRONMENT
  value: online_starter
```

Error in test:
```
    Given I have a project                                                    # features/step_definitions/project.rb:7
      please specify default environment key in config or BUSHSLICER_DEFAULT_ENVIRONMENT env variable (RuntimeError)
      /home/jenkins/bushslicer/lib/world.rb:95:in `env'
      /home/jenkins/bushslicer/lib/world.rb:147:in `user'
      /home/jenkins/bushslicer/features/step_definitions/project.rb:21:in `/^I have a project$/'
      features/cli/build.feature:6:in `Given I have a project'
```

Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Once passing the parameter variables to the container through environment variables this worked.
```
      - name: BUSHSLICER_DEFAULT_ENVIRONMENT
        value: ${BUSHSLICER_DEFAULT_ENVIRONMENT}
```

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 7 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: moved to a variable.

**Comment by kwoodson** [comment]
I have pushed an updated bushslicer image.  In the future the image needs to get updated.  

@cjryan, We need the updated test string and this should be good.

/hold

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 8 above)
Comment (kargakis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: 24h sgtm

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 10]
```diff
@@ -23,12 +23,24 @@ parameters:
 - name: DEPLOY_VERSION
   value: "v3.11"
 - name: TEST_IMAGE
-  value: "registry.svc.ci.openshift.org/azure/test-base:latest"
+  value: quay.io/kwoodson/bushslicer
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Revert this change.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 11]
```diff
@@ -23,12 +23,24 @@ parameters:
...
+  #value: "registry.svc.ci.openshift.org/azure/test-base:latest"
 - name: TEST_COMMAND
-  value: "make e2e"
+  value: "/usr/bin/scl enable rh-git29 rh-ror50 -- cucumber features/cli/build.feature:5"
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: This too

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 12]
```diff
@@ -23,12 +23,24 @@ parameters:
...
+  #value: "make e2e"
 - name: RESOURCEGROUP_SUFFIX
   generate: expression
   from: "[a-z]{8}"
+- name: TESTENV
+  value: BUSHSLICER_V3
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: You should be setting these values in the bushslicer prowjob

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 13]
```diff
@@ -126,9 +150,19 @@ objects:
...
+        if [[ "${TEST_IMAGE}" == *"/bushslicer"* ]]; then
+          TEST_CLUSTER=$(awk '/^  fqdn:/{ print $2}' /tmp/shared/openshift-azure/_data/containerservice.yaml)
+          export OPENSHIFT_ENV_ONLINE_STARTER_HOSTS="${TEST_CLUSTER}:etcd:master:node"
+          # login to cluster as $BUSHSLICER_USER
+          PASSWD=$(awk "/${BUSHSLICER_USER_PW_STRING}:/{ print \$2 }" /tmp/shared/openshift-azure/_data/containerservice.yaml)
+          oc login -u "${BUSHSLICER_USER}" -p "${PASSWD}" --server=https://${TEST_CLUSTER} --insecure-skip-tls-verify=true
+          TEST_TOKEN=$(oc whoami -t)
+          export OPENSHIFT_ENV_ONLINE_STARTER_USER_MANAGER_USERS=":${TEST_TOKEN}"
+        else
+          mkdir -p /go/src/github.com/openshift/
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: This part of the code needs a rebase.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 12 above)
Comment (mjudeikis) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: you mean jsut define `- name: TESTENV` here, without values and pass those from prow?

Comment (kargakis) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Yes, the _values_ need to come from the bushslicer prowjob.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 1 above)
Comment (charlesakalugwu) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: this image "weighs" 2.1G. thats huuugeee!!

Comment (mjudeikis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: Wanna try docker pull weblogic? :D

Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: @cjryan, I believe this could be reduced if we base the image off of centos and remove java and jenkins dependencies.  This would also require yet another dockerfile.  WDYT?

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 10 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: yeah, sorry, wasn't sure how to test this with a different value from the cli.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 11 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: same as above.  Any suggestion on how to test this with different environment variables?  Can I set them on the CLI and have them run?

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 12 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Will move to the periodic job.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 11 above)
Comment (kargakis) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Yes, if you run ci-operator locally, exposing them in your env is the same thing.

Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Thanks, will give that a shot.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 1 above)
Comment (cjryan) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: @kwoodson yeah, I agree. The issue here is the existing dockerfile started life as a jenkins node worker; since this is a ci-operator job, we could probably eliminate a lot  of the existing filesize. Also suggested this here: https://github.com/openshift/BushSlicer/pull/30#issuecomment-443021606 will see if we can get a quorum with @akostadinov

Comment (droslean) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: I have already done that in [openshift/verification-tests#30](https://github.com/openshift/verification-tests/pull/30). This is image can be used in our openshift-ci. The last think that we need to determine is if all the dependencies that are currently being installed [here](https://github.com/openshift/verification-tests/tree/master/tools/os_deps), are needed.

Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: @droslean, When your PR merges and gets verification-tests repository built then we can switch to using that image.

Comment (droslean) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: you can use the value `registry.svc.ci.openshift.org/ci/bushslicer:centos` or configure the ci-operator config `build_root` to use that image instead.

Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: @droslean, thanks!

**Comment by kwoodson** [comment]
This depends on https://github.com/openshift/verification-tests/pull/37 as we directly call it in our test image.

**Comment by droslean** [comment]
but the tests are running inside that image. Why cucumber should use it again?

**Comment by kwoodson** [comment]
@droslean, Could you clarify your statement?
> but the tests are running inside that image. Why cucumber should use it again?

If the script from openshift/verification-tests#37 does not exist in the image then we cannot execute the test suite.   Maybe I missed something.

**Comment by kwoodson** [comment]
@droslean, Could you clarify your statement?
> but the tests are running inside that image. Why cucumber should use it again?

If the script from openshift/verification-tests#37 does not exist in the image then we cannot execute the test suite.   Maybe I missed something.

**Comment by droslean** [comment]
@kwoodson In the `ci-operator` config, you can use it as the root image.  Then the cucumber will run inside that image. I am not sure why the tests need another env var to specify the image again.

**Comment by kwoodson** [comment]
@droslean, Thanks for the quick reply.  We templatize our test image and our test command so that we can re-use the same template depending on the test suite we want to run.  Therefore the image and the command are variablized.

**Comment by droslean** [comment]
@kwoodson That script will run only once, against the cluster, and after the results, the cluster will teardown.  Therefore, you won't need to run it again. The next run will run in a different cluster.

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml [hunk 14]
```diff
@@ -157,13 +175,24 @@ objects:
...
+          PASSWD=$(awk "/${BUSHSLICER_USER_PW_STRING}:/{ print \$2 }" /tmp/shared/openshift-azure/_data/containerservice.yaml)
+          oc login -u "${BUSHSLICER_USER}" -p "${PASSWD}" --server=https://${BUSHSLICER_TEST_CLUSTER} --insecure-skip-tls-verify=true
+          export BUSHSLICER_TEST_TOKEN=$(oc whoami -t)
+          export BUSHSLICER_TEST_RESULTS
+          export BUSHSLICER_TEST_FORMAT
+        else
+          mkdir -p /go/src/github.com/openshift/
+          cp -r /tmp/shared/openshift-azure /go/src/github.com/openshift/
+          cd /go/src/github.com/openshift/openshift-azure/
+          mkdir -p /go/src/github.com/openshift/openshift-azure/secrets
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: This has been removed in master.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 15]
```diff
@@ -551,3 +551,68 @@ periodics:
...
+  name: periodic-ci-azure-bushslicer
+  skip_cloning: true
+  spec:
+    containers:
+    - args:
+      - --git-ref=openshift/openshift-azure@master
+      - --secret-dir=/usr/local/e2e-secrets/azure
+      - --target=e2e-azure
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: e2e-azure-bushslicer

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 16]
```diff
@@ -551,3 +551,68 @@ periodics:
...
+      - --template=/usr/local/e2e-azure
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: --template=/usr/local/e2e-azure-bushslicer

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 17]
```diff
@@ -551,3 +551,68 @@ periodics:
     - configMap:
         name: prow-job-cluster-launch-e2e-azure
       name: job-definition
+- agent: kubernetes
+  cron: ""
+  decorate: true
+  interval: 24h
+  name: periodic-ci-azure-bushslicer
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: periodic-ci-openshift-openshift-master-azure-e2e-bushslicer

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 18]
```diff
@@ -551,3 +551,68 @@ periodics:
...
+      env:
+      - name: CLUSTER_TYPE
+        value: azure
+      - name: CONFIG_SPEC
+        valueFrom:
+          configMapKeyRef:
+            key: openshift-openshift-azure-master.yaml
+            name: ci-operator-configs
+      - name: JOB_NAME_SAFE
+        value: e2e-azure
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: e2e-azure-bushslicer

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 19]
```diff
@@ -551,3 +551,68 @@ periodics:
...
+      name: ""
+      resources:
+        limits:
+          cpu: 500m
+        requests:
+          cpu: 10m
+      volumeMounts:
+      - mountPath: /usr/local/e2e-secrets/azure
+        name: cluster-secrets-azure
+      - mountPath: /usr/local/e2e-azure
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: e2e-azure-bushslicer

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml (hunk 14 above)
Comment (kwoodson) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure.yaml]: Will rebase as noted by gh.  Thanks.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 15 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: Will update.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 16 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: Will update.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 17 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: updated.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 18 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: updated.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml (hunk 19 above)
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: Updated.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 20]
```diff
@@ -605,3 +605,68 @@ periodics:
...
+  decorate: true
+  interval: 24h
+  name: periodic-ci-openshift-openshift-master-azure-e2e-bushslicer
+  skip_cloning: true
+  spec:
+    containers:
+    - args:
+      - --git-ref=openshift/openshift-azure@master
+      - --secret-dir=/usr/local/e2e-secrets/azure
+      - --target=e2e-azure-bushslicer
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: I think you are missing a couple of other targets here for building the rest of the osa images (sync, azure-controllers, etcdbackup). See `periodic-ci-openshift-openshift-azure-conformance` as an example.

Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: What happens if these aren't created?  Can it pull latest?  Are these required to be built each time?  I will add them if required.

Comment (kargakis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: Good questions, they may not be required for periodics, let's merge and see it if breaks.

Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: I have updated the PR to reflect that we want sync and controllers.  I pass those locally so I've updated the job here with those parameters.

**Comment by kwoodson** [comment]
Squashed to avoid the daily rebase conflicts.
